test/04_calendar_date_overrides.js

```diff
@@ -37,48 +37,170 @@ var Titles = Romcal.Titles;
 var Types = Romcal.Types;
 var PsalterWeeks = Romcal.PsalterWeeks;
 var Calendar = Romcal.Calendar;
+var CalendarsDef = require('../src/calendars');
 
 describe('Testing national calendar overrides', function() {
 
   this.timeout(0);
 
+  describe('An optional celebration is available to be celebrated, in addition to the feria', function() {
+    var generalDates2020 = Calendar.calendarFor(2020);
+    var generalDates2021 = Calendar.calendarFor(2021);
+    var spainDates2020 = Calendar.calendarFor({ year: 2020, country: 'spain' });
+    it('The optional memory of the Most Holy Name of Jesus is available on the 3th of January, in addition to the feria', function() {
+      var dates = _.filter(generalDates2020, function(d) {
+        return d.moment.isSame(moment.utc({ year: 2020, month: 0, day: 3 }));
+      });
+      _.size(dates).should.be.eql(2);
+    });
+    it('However, if the 3th of January is a Sunday, the Solemnity of Epiphany takes the precedence.', function() {
```

Comment:
I hope you count in this test with the option that the Epiphany might be celebrated on 6 January. I believe that in the GRC, Epiphany should be always celebrated on 6 January.

Reply:
This test run in the context of the year `2020`, from the `general` calendar, with the default `epiphanyOnJan6` set to `false`. Everything is scoped.